mathlibtools/lib.py

```diff
@@ -842,12 +842,20 @@ def reduce_imports_sed(self, file: str) -> Iterable[str]:
 
     def make_all(self) -> None:
         """Creates all.lean importing everything from the project"""
+
+        # helper function to wrap file paths in double quotes if they
+        # contain dashes, so that lean accepts them as imports
+        def make_safe(s):
+            if "-" in s:
+                return "«" + s + "»"
+            return s
```

Comment:
Does it make sense to use the regex from here?

https://github.com/leanprover/vscode-lean/blob/2b43982c4c6305a0f20f156152a60613a6f1a683/syntaxes/lean.json#L184-L186

Reply:
I'm not sure what the rules are for imports, and even a regex for the valid names wouldn't tell us which ones need escaping. I considered adding `.` here also, but I don't know if anyone would even try to use a `.` in a lean file path?

Reply:
It looks like import tokens are no different to regular identifiers:

https://github.com/leanprover-community/lean/blob/4f0963ee6cc8380b0178fa9930797f0cfb730694/src/frontends/lean/parser.cpp#L2682

https://github.com/leanprover-community/lean/blob/4f0963ee6cc8380b0178fa9930797f0cfb730694/src/frontends/lean/parser.cpp#L2711-L2712

Which suggests the above regex is the right one.

Reply:
For reference, this is how lean itself decides when to escape identifiers:
```
https://github.com/leanprover-community/lean/blob/65ad4ffdb3abac75be748554e3cbe990fb1c6500/src/util/name.cpp#L65-L83
```

Reply:
Oh nice detective work! So the regex you posted is the right one! Wow I'm surprised that so much stuff doensn't need escaping I guess